c_src/elmdb_nif.c

```diff
@@ -452,7 +453,10 @@ static int to_mdb_cursor_op(ErlNifEnv *env, ERL_NIF_TERM op, MDB_val *key) {
   return 0;
 }
 
-static ElmdbEnv* open_env(const char *path, int mapsize, int maxdbs, int envflags, int *ret) {
+/* FIXME (DD 26/8/2018) this function should return some
+ * sensible error codes so that callers in erlang know what is going on.
+ * Plus just exiting the erlang thread blindly is causing the BEAM to segfault.*/
+static ElmdbEnv* open_env(const char *path, mdb_size_t mapsize, int maxdbs, int envflags, int *ret) {
```

Comment:
Changing the type for `mapsize` here to `mdb_size_t` fixes a problem where databases beyond a certain threshold fails (1GB? IIRC).